### PR TITLE
Try to make a new class to handle external calls

### PR DESCRIPTION
**Pull request by dangershony**
This is a failed attempt due to my knowledge of angular being limited.
The idea is to create  anew class called `CommunicationExternal` that will handle external calls.
However I could not get the `ext:login` to get triggered

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -91,3 +91,4 @@ node_modules/
 
 # Yarn Integrity file
 .yarn-integrity
+/.vs
```
Comment (sondreb) [inline .gitignore:94]: Do you use Visual Studio 2022 for this? Get Visual Studio Code :-)

File: angular/src/app/services/application-manager.ts [hunk 2]
```diff
@@ -16,7 +16,8 @@ export class AppManager {
         public sync: DataSyncService,
         public state: StateService,
         public walletManager: WalletManager,
-        public communication: CommunicationService,
+      public communication: CommunicationService,
+      public communicationExternal: CommunicationExternal,
```
Comment (sondreb) [inline angular/src/app/services/application-manager.ts:20]: Good call on having a separated service for this, makes it cleaner.

File: angular/src/manifest/blockcore/manifest.json [hunk 3]
```diff
@@ -1,44 +1,54 @@
...
+  },
+  "externally_connectable": {
+    "matches": [
+      "https://www.blockcore.net/*",
+      "http://localhost/*"
+    ]
+  },
+  "web_accessible_resources": [
+    {
+      "resources": [ "blockcore-loader.js", "provider.js" ],
```
Comment (sondreb) [inline angular/src/manifest/blockcore/manifest.json:46]: The "blockcore-loader" should not be added here.

File: extension/src/background.ts [hunk 4]
```diff
@@ -67,27 +73,57 @@ chrome.alarms.onAlarm.addListener(async (alarm: chrome.alarms.Alarm) => {
...
+    }
+
+    var wallets = walletStore.getWallets();
+
+    var res: any;
+    for (let i = 0; i < wallets.length; i++) {
+      const wallet = wallets[i];
+      for (let j = 0; j < wallet.accounts.length; j++) {
+        const account = wallet.accounts[j];
+        res = account.xpub
```
Comment (sondreb) [inline extension/src/background.ts:102]: Never share xpub outside of the wallet.

**Review by sondreb — CHANGES_REQUESTED** [review]
Thanks, added some comments if you want to look at them, at least let's remove the xpub issue before we merge this.

File: .gitignore (hunk 1 above)
Comment (dangershony) [reply .gitignore:94]: yeah I don't know for some reason I feel more comfortable with vs, I will try code as well

File: extension/src/background.ts (hunk 4 above)
Comment (dangershony) [reply extension/src/background.ts:102]: yeah sure, I didn't have access to the hd class so this is just an example

File: angular/src/app/services/communication.external.ts [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        });
+      }
+    }
+
+  getNetwork(networkType: string) {
+    return this.allNetworks.find(w => w.id == networkType);
+  }
+
+
+  async handleExternalMessage(message: Message, sender: chrome.runtime.MessageSender) {
```
Comment (dangershony) [inline angular/src/app/services/communication.external.ts:64]: I was not able to get this method to trigger @sondreb maybe you can find why?

File: angular/src/manifest/blockcore/manifest.json [hunk 6]
```diff
@@ -1,44 +1,54 @@
...
-                "content.js"
-            ],
-            "matches": [
-                "<all_urls>"
-            ]
-        }
-    ],
-    "content_security_policy": {
-        "extension_pages": "script-src 'self'; object-src 'self'"
+  "name": "Blockcore Wallet",
```
Comment (dangershony) [inline angular/src/manifest/blockcore/manifest.json:2]: why do my ide remove spaces? maybe  am using tabs?

**Review by sondreb — APPROVED** [review]
Great work, thanks!

**Pull request merged**